### ISSUE TITLE
prefer-top-level-await flags code that can't be top-level-awaited

### ISSUE THREAD

**Issue by VlaDexa** [sindresorhus/eslint-plugin-unicorn#1860]
In class definitions top-level-await is not available but this rule still marks the code as wrong

The rule: [unicorn/prefer-top-level-await](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v43.0.0/docs/rules/prefer-top-level-await.md)

```js
class Example {
    property = Promise // Can't await here
        .resolve(1)
        .then(i => i+1); // Error here <- Prefer top-level await over using a promise chain.
}
```

**Comment by fisker** [comment]
I'm confused, I can't find anything forbid TLA in class member, [the spec](https://tc39.es/proposal-top-level-await/), [acorn PR](https://github.com/acornjs/acorn/commit/65a9137ac9501d27e41aa37a48e1eddc0c605cda), [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#top_level_await). So the question is where the TLA expression can be?

**Comment by fisker** [comment]
Because when parsing class member, the parser enters a non-await function? Need more time to learn the spec.

**Comment by VlaDexa** [comment]
Maybe the Initializer which runs for ClassFields is a part of the constructor which doesn't allow async?
